Podcast/SearchViewController.swift

```diff
@@ -72,35 +218,101 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
+                pastSearchesTableView.isHidden = false
+                pastSearchesTableViewReloadData()
+                mainScrollView = pastSearchesTableView
+            }
+            return
+        }
+        tabUnderlineView.isHidden = false
+        searchResultsTableView.isHidden = false
+        pastSearchesTableView.isHidden = true
+        didDismissItunesHeaderForQuery = false
+        addSearchITunesHeader()
+        mainScrollView = searchResultsTableView
+        searchResultsTableView.startLoadingAnimation()
+        searchResultsTableView.loadingAnimation.bringSubview(toFront: searchResultsTableView)
+        for data in tableViewData {
+            data.fetchData(query: searchText, newSearch: true)
+        }
+        searchResultsTableView.reloadData()
+        self.updateTableViewInsetsForAccessoryView()
+    }
+
+    // MARK: - Tab Bar Delegate
+
+    func selectedTabDidChange(toNewIndex newIndex: Int) {
+        addSearchITunesHeader()
+        searchResultsTableView.dataSource = tableViewData[newIndex]
+        searchResultsTableView.delegate = tableViewData[newIndex]
+        if let data = searchResultsTableView.dataSource as? SearchDataSourceDelegate, data.completingNewSearch {
+            searchResultsTableView.startLoadingAnimation()
+        } else {
+            searchResultsTableView.stopLoadingAnimation()
+        }
+        searchResultsTableView.reloadData()
+    }
+
+    // MARK: - SearchItunesHeaderViewDelegate
+
+    func searchITunesHeaderDidPressSearchITunes(searchITunesHeader: SearchITunesHeaderView) {
+        let searchItunesViewController = SearchITunesViewController(query: searchController.searchBar.text ?? "")
+        navigationController?.pushViewController(searchItunesViewController, animated: true)
+    }
+
+    func searchITunesHeaderDidPressDismiss(searchITunesHeader: SearchITunesHeaderView) {
+        searchResultsTableView.tableHeaderView = nil
+        didDismissItunesHeaderForQuery = true
+    }
+
     //MARK: - Tabbed Search Results Delegate
-    func didTapOnSeriesCell(series: Series) {
+
+    func refreshController() {
+        searchResultsTableView.finishInfiniteScroll()
+        searchResultsTableView.reloadData()
+        searchResultsTableView.stopLoadingAnimation()
+        self.updateTableViewInsetsForAccessoryView()
+    }
+
+    func didSelectCell(cell: UITableViewCell, object: Any) {
+        if let series = object as? Series {
+            didTapOnSeriesCell(series: series)
+        } else if let episode = object as? Episode {
+            didTapOnEpisodeCell(episode: episode)
+        } else if let user = object as? User {
+            didTapOnUserCell(user: user)
+        }
+    }
+
+    private func didTapOnSeriesCell(series: Series) {
         addPastSearches()
         let seriesDetailViewController = SeriesDetailViewController(series: series)
         navigationController?.pushViewController(seriesDetailViewController,animated: true)
```

Comment:
![image](https://user-images.githubusercontent.com/20246620/34462904-611726aa-ee1b-11e7-958e-48346acdd99e.png)
going back from a series detail/episode detail to the search causes the search results to disappear for a few seconds/not sure why this is happening

Reply:
I can't reproduce this.. any more details?

Reply:
nevermind fixed.